tag-system/src/main/java/org/togetherjava/tjbot/tag/TagSystemInstanceHolder.java

```diff
@@ -0,0 +1,33 @@
+package org.togetherjava.tjbot.tag;
+
+import org.togetherjava.tjbot.db.Database;
+
+public class TagSystemInstanceHolder {
+    private TagSystemInstanceHolder() {}
+
+    private static volatile TagSystem instance;
+
+    public static TagSystem getOrCreate(Database database) {
+        if (instance == null) {
+            setInstance(new TagSystem(database));
+        }
+
+        return instance;
+    }
+
+    public static TagSystem getInstance() {
+        if (instance == null) {
+            throw new UnsupportedOperationException("Instance not yet set");
```

Comment:
this is the wrong exception imo, since the operation **is supported**, just not with the current state
```suggestion
            throw new IllegalStateException("Instance not yet set");
```

Reply:
done